win32/tests/job_object_helper_int/job_object_helper_int.c

```diff
@@ -0,0 +1,170 @@
+// Copyright(C) Microsoft Corporation.All rights reserved.
+
+
+#include <stdlib.h>
+
+#include "windows.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+
+#include "c_pal/timer.h"
+#include "c_pal/job_object_helper.h"
+
+#define MEGABYTE ((size_t)1024 * 1024)
+#define GIGABYTE (MEGABYTE * 1024)
+#define MAX_MEMORY_POSSIBLE (GIGABYTE * 256)
+#define TEST_BUFFER_SIZE (MEGABYTE * 512)
+#define MAX_BUFFERS_BEFORE_FAILURE (MAX_MEMORY_POSSIBLE / TEST_BUFFER_SIZE)
+
+static const size_t max_buffers_before_failure = MAX_BUFFERS_BEFORE_FAILURE;
+
+static size_t get_malloc_limit_before_failure(void)
+{
+    /* malloc until failure and return the count of (512MB) buffers we were able to malloc. */
+    void** buffers = malloc(sizeof(void*) * max_buffers_before_failure);
+    ASSERT_IS_NOT_NULL(buffers);
+
+    size_t max_buffers = 0;
+    for (size_t i = 0; i < max_buffers_before_failure; i++)
+    {
+        buffers[i] = malloc(TEST_BUFFER_SIZE);
+        if (buffers[i] == NULL)
+        {
+            max_buffers = i;
+            break;
+        }
+    }
+    ASSERT_ARE_NOT_EQUAL(size_t, 0, max_buffers);
+    for (size_t i = 0; i < max_buffers; i++)
+    {
+        free(buffers[i]);
+    }
+    free(buffers);
+
+    return max_buffers;
+}
+
+
+/* We need something to keep the CPU busy. */
+/* If you list the numbers from 0 to 2^28-1 in binary, you'll find that 3,758,096,384 bits are set. */
+/* Lets verify this. It takes about 5 seconds on my desktop PC. */
+static size_t limit = 1 << 28;
+static size_t expected_bits = 3758096384;
+
+static size_t get_elapsed_milliseconds_for_cpu_bound_task(void)
+{
+    double start = timer_global_get_elapsed_ms();
+
+    size_t total_bits = 0;
+
+    for (size_t i = 0; i < limit; i++)
+    {
+        size_t this = i;
+        while (this != 0)
+        {
+            total_bits += 1;
+            this = this & (this - 1);
```

Comment:
>this [](http://example.com/codeflow?start=12&length=4)

Can we please have this variable named to anything else which is not a keyword in C++?...